apps/trading/pages/portfolio/withdrawals/withdrawals-list.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  getDateTimeFormat,
+  t,
+  truncateByChars,
+  formatNumber,
+} from '@vegaprotocol/react-helpers';
+import { WithdrawalStatus } from '@vegaprotocol/types';
+import {
+  EtherscanLink,
+  AgGridDynamic as AgGrid,
+} from '@vegaprotocol/ui-toolkit';
+import { EthTxStatus, TransactionDialog } from '@vegaprotocol/web3';
+import { useCompleteWithdraw } from '@vegaprotocol/withdraws';
+import type { ValueFormatterParams } from 'ag-grid-community';
+import { AgGridColumn } from 'ag-grid-react';
+import orderBy from 'lodash/orderBy';
+import { useEffect, useMemo } from 'react';
+import type { WithdrawalsPageQuery_party_withdrawals } from './__generated__/WithdrawalsPageQuery';
+
+interface WithdrawalsListProps {
+  withdrawals: WithdrawalsPageQuery_party_withdrawals[];
+  refetchWithdrawals: () => void;
+}
+
+export const WithdrawalsList = ({
+  withdrawals,
+  refetchWithdrawals,
+}: WithdrawalsListProps) => {
+  const { transaction, submit } = useCompleteWithdraw();
+
+  const sortedWithdrawals = useMemo(() => {
+    return orderBy(
+      withdrawals,
+      (w) => new Date(w.createdTimestamp).getTime(),
+      'desc'
+    );
+  }, [withdrawals]);
+
+  // TODO: Get this working, sometimes the table doesnt update because we have to
```

Comment:
Sounds like this needs a separate ticket

Reply:
I'll take a look. I think what we can do is, rather than refetchWithdrawals we start a bus event sub in the hook listening for the matching withdrawal, as soon as we see it _and_ it has a foreign chain tx hash we can update it